treadmill-scripts/newldap.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+#########################
+# The command line help #
+#########################
+display_help() {
+    echo "Usage: $0 [option...]" >&2
+    echo
+    echo "   -s, --subnet           AWS subnet IDs; e.g. -s subnet-1234 -s subnet-5678 -s subnet-9999"
+    echo "   -a, --ami              AWS AMI image; e.g. ami-1234"
+    echo "   -c, --cell             New Treadmill cell name; e.g. userdev"
+    echo "   -d, --domain           IPA domain; e.g. ipa.foo.com"
+    echo "   -n, --number           Number of LDAP servers, e.g. 2"
+    echo "   -l, --ldapsuffix       LDAP suffix; e.g. dc=foo,dc=com"
+    echo "   -r, --realm            Realm; e.g. FOO.COM"
+    echo "   -p, --proid            LDAP auth user; e.g admin"
+    echo
+    exit 0
+}
+SUBNETS=()
+NUMBER=0
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    -s|--subnet)
+    SUBNETS=("${SUBNETS[@]}" "$2")
+    shift # past argument
+    shift # past value
+    ;;
+    -a|--ami)
+    AMI="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -c|--cell)
+    TREADMILL_CELL="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -d|--domain)
+    TREADMILL_DNS_DOMAIN="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -n|--number)
+    NUMBER="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -l|--ldapsuffix)
+    TREADMILL_LDAP_SUFFIX="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -r|--realm)
+    REALM="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -p|--proid)
+    REALM="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -h|--help)
+    display_help
+    ;;
+esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+if [ ! ${#SUBNETS[@]} -eq $NUMBER ]; then
+	echo -e "Please pass exactly the same number of subnets as requested number of servers; exiting."
+	exit 1
+fi
+
+echo 
+echo "AWS SUBNETS       = ${SUBNETS[@]}"
+echo "AWS AMI           = ${AMI}"
+echo "TREADMILL CELL    = ${TREADMILL_CELL}"
```

Comment:
you do not need cell for ldap. ldap is cross cell.

Reply:
Removed, thanks.